src/components/views/ShareEcash.tsx

```diff
@@ -0,0 +1,126 @@
+import CustomCarousel from '../utility/Carousel/CustomCarousel';
+import ClipboardButton from '../buttons/utility/ClipboardButton';
+import StickerItem from '../eGifts/stickers/StickerItem';
+import { useEffect, useMemo, useState } from 'react';
+import ErrorBoundary from '../utility/ErrorBoundary';
+import AnimatedQRCode from '../utility/QR/AnimatedQR';
+import { GiftAsset, PublicContact } from '@/types';
+import QRCode from 'qrcode.react';
+import UserLink from '../utility/UserLink';
+
+interface ShareEcashProps {
+   gift?: GiftAsset;
+   token?: string;
+   txid?: string;
+   contact?: PublicContact;
+   onClose: () => void;
+}
+
+const ShareEcash = ({ token, txid, gift, onClose, contact }: ShareEcashProps) => {
+   const [carouselSlides, setCarouselSlides] = useState<React.ReactNode[]>([]);
+
+   useEffect(() => {
```

Comment:
btw this unnecessary useEffect. you could do something like this intead:
```
const carouselSlides = useMemo(() => {
      if (!token) return [];

      const qrCodeValue = `${window.location.protocol}//${window.location.host}/wallet?token=${token}`;
      
      return [
         <div className='text-black text-center space-y-2 ml-3' key='1'>
            <ErrorBoundary
               fallback={
                  <p className='text-red-500'>
                     Failed to load QR. Link too long. Go to transaction history to reclaim your
                     tokens and try again, or just copy the link by clicking the button below.{' '}
                  </p>
               }
            >
               <QRCode value={qrCodeValue} size={275} />
               <p> Link: {`boardwalkcash.com...`}</p>
            </ErrorBoundary>
         </div>,
         <div className='text-black text-center space-y-2' key='2'>
            <ErrorBoundary
               fallback={<p className='text-red-500'>Failed to load QR. Token too long</p>}
            >
               <AnimatedQRCode encodedToken={`${token}`} />
            </ErrorBoundary>
            <p> Token: {`${token.slice(0, 12)}...${token.slice(-5)}`}</p>
         </div>,
      ];
   }, [token]);
   ```
   
   maybe even without useMemo